### PR TITLE
update rubocop to 0.41.2 except one possible buggy part, that need ve…

### PR DESCRIPTION
**Pull request by jreidinger**
…rification

remaining problem was introduced by https://github.com/yast/yast-network/commit/095e1dbcad84a0075da6371bc57657506a19b327

which looks like bug

### CODE REVIEW

File: src/clients/routing.rb [hunk 1]
```diff
@@ -402,11 +403,9 @@ def DeleteHandler(options)
       options = deep_copy(options)
       found = false
       Routing.Routes = Builtins.maplist(Routing.Routes) do |m|
-        if Ops.get(m, "destination") == Ops.get(options, "dest")
-          found = true
-        else
-          next deep_copy(m)
-        end
+        next deep_copy(m) if Ops.get(m, "destination") == Ops.get(options, "dest")
```
Comment (mchf) [inline src/clients/routing.rb]: should be ```next deep_copy(m) if Ops.get(m, "destination") != Ops.get(options, "dest")```, shouldn't it?

File: src/include/network/complex.rb [hunk 2]
```diff
@@ -378,33 +370,29 @@ def DeviceStatus(devtype, devname, devmap)
...
-            )
+            return Builtins.sformat(_("Configured as %1 with address %2"), devname, addr)
           end
-        else
-          # Network card status (%1 is device, %2 is protocol)
-          return Builtins.sformat(
-            _("Configured as %1 with %2"),
+
+          # Network card status (%1 is device, %2 is address, %3 is address)
+          Builtins.sformat(
```
Comment (mchf) [inline src/include/network/complex.rb]: missing ```return```

File: test/network_autoconfiguration_test.rb [hunk 3]
```diff
@@ -66,10 +66,10 @@ def probe_netcard_factory(num)
 
     allow(Yast::SCR).to receive(:Read) do |path|
       if path.to_s =~ /^\.network\.value\."(eth\d+)".(.*)/
-        ifcfg_files.get(Regexp.last_match(1), Regexp.last_match(2))
-      else
-        raise "Unexpected Read #{path}"
+        next ifcfg_files.get(Regexp.last_match(1), Regexp.last_match(2))
```
Comment (mchf) [inline test/network_autoconfiguration_test.rb]: I didn't get this

File: test/default_route_test.rb [hunk 4]
```diff
@@ -29,10 +29,10 @@
 
     allow(Yast::SCR).to receive(:Read) do |path|
       if path.to_s =~ /^\.network\.value\."(eth\d+)".(.*)/
-        @ifcfg_files.get(Regexp.last_match(1), Regexp.last_match(2))
-      else
-        raise "Unexpected Read #{path}"
+        next @ifcfg_files.get(Regexp.last_match(1), Regexp.last_match(2))
```
Comment (mchf) [inline test/default_route_test.rb]: i didn't get this

File: src/clients/routing.rb (hunk 1 above)
Comment (jreidinger) [inline src/clients/routing.rb]: you are right, will fix it

File: src/include/network/complex.rb (hunk 2 above)
Comment (jreidinger) [inline src/include/network/complex.rb]: good catch

File: test/default_route_test.rb (hunk 4 above)
Comment (jreidinger) [inline test/default_route_test.rb]: in block next is in its context like local return. So this next do quick return of this value.

It is similar like if you have method
```
def m
  if a
    return val
  elsif b
     return valb
  end

  raise "unknown value"
end
```

**Comment by mchf** [comment]
LGTM

**Comment by jreidinger** [comment]
thanks

**Pull request merged**